firebase-firestore/src/main/java/com/google/firebase/firestore/core/UserData.java

```diff
@@ -303,9 +301,15 @@ private void validatePath() {
       }
     }
 
+    private static final String RESERVED = "__";
```

Comment:
I don't feel strongly but is there a reason to not use the regex anymore?  JavaScript at least still uses the regex, so this represents a minor porting difference...  and the regex is a bit more self-explanatory, since RESERVED is only reserved if it occurs at the beginning and the end (if you keep it as-is, perhaps add a comment)...

Reply:
Javascript uses the regex, but iOS doesn't: https://github.com/firebase/firebase-ios-sdk/blob/master/Firestore/core/src/firebase/firestore/core/user_data.mm#L178

Renamed to match iOS even more closely: `RESERVED_FIELD_DESIGNATOR`